### ISSUE TITLE
DateTimeImmutable::setTime incorrect return value

### ISSUE THREAD

**Issue by rbairwell** [kalessil/phpinspectionsea#1504]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate) 3.0.7 |
| Language level | PHP 7.4                   |

Current behaviour
--
The inspection `[EA] New value type (bool) is not in annotated types. ` is flagged on `->setTime` line the below code example:
```php
<?php
declare(strict_types=1);

function test(\DateTimeImmutable $time) : \DateTimeImmutable
{
$time=$time->setTime(23,45,23);
return $time;
}
$oldtime=new \DateTimeImmutable();
print test($oldtime)->format('c');
```

Expected behaviour
--
No inspection should be flagged as https://www.php.net/manual/en/datetimeimmutable.settime.php returns a new DateTimeImmutable object.

Environment details
---
PhpStorm 2019.3.1
Build #PS-193.5662.63, built on December 19, 2019
Runtime version: 11.0.5+10-b520.17 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1981M
Cores: 8
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: GrepConsole, Key Promoter X, NEON support, PHPUnit code coverage, String Manipulation, com.intellij.ideolog, com.intellij.plugin.adernov.powershell, com.kalessil.phpStorm.phpInspectionsUltimate, de.espend.idea.laravel, de.espend.idea.php.toolbox, de.espend.idea.php.annotation, de.espend.idea.php.phpunit, izhangzhihao.rainbow.brackets, net.seesharpsoft.intellij.plugins.csv, org.jetbrains.plugins.rest, org.psliwa.idea.composer, ru.adelf.idea.dotenv

**Comment by andrew-demb** [comment]
See https://www.php.net/manual/en/datetime.settime.php#refsect1-datetime.settime-returnvalues

Method can return `false` on failure

**Comment by rbairwell** [comment]
Thanks @andrew-demb - I missed that.

I can confirm:
```php
<?php
declare(strict_types=1);

function test(\DateTimeImmutable $time): \DateTimeImmutable
{
    $new = $time->setTime(23, 45, 23);
    if ($new === false) {
        throw new \InvalidArgumentException('Unable to set time');
    }
    return $time;
}

$oldtime = new \DateTimeImmutable();
print test($oldtime)->format('c');
```
works correctly.
